R/ResultFlagsDependent.R

```diff
@@ -76,39 +76,43 @@ InvalidFraction <- function(.data, clean = TRUE, errorsonly = FALSE) {
   # rename NA values to Nonstandardized in WQX.SampleFractionValidity column
   check.data["WQX.SampleFractionValidity"][is.na(check.data["WQX.SampleFractionValidity"])] <- "Nonstandardized"
   
-  # reorder column names to match .data
-  # get .data column names
-  col.order <- colnames(.data)
-  # add WQX.SampleFractionValidity column to the list
-  col.order <- append(col.order, "WQX.SampleFractionValidity")
-  # reorder columns in check.data
-  check.data <- check.data[, col.order]
-  # place flag column next to relevant fields
-  check.data <- check.data %>%
-    dplyr::relocate("WQX.SampleFractionValidity",
-                    .after = "ResultSampleFractionText"
-    )
+  # EDH replaced with function OrderTADACols at end
+  # # reorder column names to match .data
+  # # get .data column names
+  # col.order <- colnames(.data)
+  # # add WQX.SampleFractionValidity column to the list
+  # col.order <- append(col.order, "WQX.SampleFractionValidity")
+  # # reorder columns in check.data
+  # check.data <- check.data[, col.order]
+  # # place flag column next to relevant fields
+  # check.data <- check.data %>%
+  #   dplyr::relocate("WQX.SampleFractionValidity",
+  #                   .after = "ResultSampleFractionText"
+  #   )
   
   # if all rows are "Valid", return input unchanged
   if (any(c("Nonstandardized", "Invalid") %in%
           unique(check.data$WQX.SampleFractionValidity)) == FALSE) {
     if (errorsonly == FALSE) {
       print("No changes were made, because we did not find any invalid fraction/characteristic combinations in your dataframe")
+      .data = OrderTADACols(.data)
       return(.data)
     }
     if (errorsonly == TRUE) {
       print("This dataframe is empty because we did not find any invalid fraction/characteristic combinations in your dataframe")
       empty.data <- dplyr::filter(check.data, WQX.SampleFractionValidity == "Invalid")
       empty.data <- dplyr::select(empty.data, -WQX.SampleFractionValidity)
+      empty.data = OrderTADACols(empty.data)
```

Comment:
Let's discuss the logic for all these flag functions: 

When flagging: If the input dataframe does not have an issues that require flagging, then I believe the function returns the input dataframe unchanged, with a note saying that. In that case, the output would not include the flag column since it would contain only NAs. For tracking purposes, do you think it would make sense to still include the flag column in the output, even if it would have all NA's (no flags)?

When flagging "errors only", it works a little differently. In that case the function will return an empty dataframe (also with no flag column).